internal/common/rules/globalrulescache.go

```diff
@@ -56,7 +56,44 @@ func pullRuleOutOfPlayInternal(rulename string, allRules map[string]RuleMatrix,
 			delete(playRules, rulename)
 			return nil
 		}
-		return errors.Errorf("Rule '%v' is not in play", rulename)
+		return &RuleError{err: errors.Errorf("Rule '%v' is not in play", rulename), errorType: RuleIsNotInPlay}
+	}
+	return &RuleError{err: errors.Errorf("Rule '%v' does not exist in available rules cache", rulename), errorType: RuleNotInAvailableRulesCache}
+}
+
+func ModifyRule(rulename string, newMatrix mat.Dense, newAuxiliary mat.VecDense) error {
```

Comment:
would be good to document as per linter as this is the API

Reply:
Agreed and added